src/lib/utils/errors.ts

```diff
@@ -279,7 +278,12 @@ function isUserNotEnoughGas(error): boolean {
  * Checks if error is caused by user's wallet having bad config / state
  */
 function isWalletConfigError(error): boolean {
-  const messages = [/invalid rpc url/, /nonce has already been used/];
+  const messages = [
+    /invalid rpc url/,
+    /nonce has already been used/,
+    /no matching key/, //Wallet connect v2 random error when disconnecting: https://github.com/WalletConnect/walletconnect-monorepo/issues/2326#issuecomment-1633706820
```

Comment:
Also moved this message from `isUserRejected` to `isWalletConfigError` as it makes more sense here.